src/libraries/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Certificate2.cs

```diff
@@ -628,6 +632,347 @@ public bool Verify()
+            ReadOnlySpan<char> keyContents = keyPemFilePath is null ? certContents : File.ReadAllText(keyPemFilePath);
+
+            return CreateFromPem(certContents, keyContents);
+        }
+
+        /// <summary>
+        /// Creates a new X509 certificate from the file contents of an RFC 7468 PEM-encoded
+        /// certificate and password protected private key.
+        /// </summary>
+        /// <param name="certPemFilePath">The path for the PEM-encoded X509 certificate.</param>
+        /// <param name="keyPemFilePath">
+        /// If specified, the path for the password protected PEM-encoded private key.
+        /// If unspecified, the <paramref name="certPemFilePath" /> file will be used to load
+        /// the private key.
+        /// </param>
+        /// <param name="password">The password for the encrypted PEM.</param>
+        /// <returns>A new certificate with the private key.</returns>
+        /// <exception cref="ArgumentException">
+        /// <para>
+        ///   The contents of the file path in <paramref name="certPemFilePath" /> do not contain
+        ///   a PEM-encoded certificate, or it is malformed.
+        /// </para>
+        /// <para>-or-</para>
+        /// <para>
+        ///   The contents of the file path in <paramref name="keyPemFilePath" /> do not contain a
+        ///   password protected PEM-encoded private key, or it is malformed.
+        /// </para>
+        /// <para>-or-</para>
+        /// <para>
+        ///   The contents of the file path in <paramref name="keyPemFilePath" /> contains
+        ///   a key that does not match the public key in the certificate.
+        /// </para>
+        /// </exception>
+        /// <exception cref="ArgumentNullException">
+        /// The path specified in <paramref name="certPemFilePath" /> is null.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        /// <para>The certificate uses an unknown public key algorithm.</para>
+        /// <para>-or-</para>
+        /// <para>The password specified for the private key is incorrect.</para>
+        /// </exception>
+        /// <remarks>
+        /// <para>
+        /// See <see cref="System.IO.File.ReadAllText(string)" /> for additional documentation about
+        /// exceptions that can be thrown.
+        /// </para>
+        /// <para>
+        /// Password protected PEM-encoded keys are always expected to have the PEM label "ENCRYPTED PRIVATE KEY".
+        /// </para>
+        /// <para>PEM-encoded items that have a different label are ignored.</para>
+        /// <para>
+        ///   Combined PEM-encoded certificates and keys do not require a specific order. For the certificate, the
+        ///   the first certificate with a CERTIFICATE label is loaded. For the private key, the first private
+        ///   key with the label "ENCRYPTED PRIVATE KEY" is loaded. More advanced scenarios for loading certificates and
```

Comment:
I think that for AsymmetricAlgorithm.ImportFromPem we landed on "if it's ambiguous at all, throw" but here it's "first".  I guess the consistency can be restablished from "because it has to coherently match the certificate", but I'm just calling out the disparity and wondering how that makes you feel :smile:.

Reply:
>  think that for AsymmetricAlgorithm.ImportFromPem we landed on "if it's ambiguous at all, throw" but here it's "first"

Right.

> wondering how that makes you feel

~"Tired" is overwhelmingly how I always feel.~ Ideally there is no possibility of ambiguity. Originally when I started this, I started this as "load the first key whose public key matches the certificate's SPKI". Perhaps that is worth implementing so that "first" really becomes "just the right one".

Reply:
I wouldn't do the "first one that works", since that arguably requires an N*M matching to find sanity.

Any of

* First with any known key label (will then fail if the wrong algorithm's algorithm-specific key is first)
* Only key with any known key label (fail if there's a second one of any kind)
* First applicable
* Only applicable

seem fine to me.  So really it's just "for keys alone it was only-applicable, for cert+key it's first-applicable/first-applicable".

Since the X509Certificate2 ctor will load the first applicable thing out of a multi-PEM, first-applicable/first-applicable might make more sense here in these classes... I'm just making sure it's something we thought about instead of something we just ended up with.

Reply:
> I wouldn't do the "first one that works", since that arguably requires an N*M matching to find sanity

Ah that wasn't quite what I thought of, I thought "first certificate always, keep looking through keys until we find one that matches", so "1 * M".

> I'm just making sure it's something we thought about instead of something we just ended up with.

Yeah ultimately I got rid of that matching logic since it was non-trivial where I think the current behavior is acceptable.